src/ducks/reimbursements/Reimbursements.jsx

```diff
@@ -1,192 +1,139 @@
-                className={styles.Reimbursements__transactionsList}
-                showTriggerErrors={false}
+    <TransactionActionsProvider>
+      <div
+        className={`${styles.Reimbursements} ${DESKTOP_SCROLLING_ELEMENT_CLASSNAME}`}
+      >
+        <Section>
+          <SectionTitle>
+            {t('Reimbursements.pending')}
+            <Figure
+              symbol="€"
+              total={pendingAmount}
+              className={styles.Reimbursements__figure}
+              signed
+            />{' '}
+          </SectionTitle>
+          {hasPendingExpenses ? (
+            <TransactionList
+              transactions={pendingExpenses}
+              className={styles.Reimbursements__transactionsList}
+              showTriggerErrors={false}
+            />
+          ) : (
+            <>
+              <Divider />
+              <NoPendingReimbursements
+                period={formattedPeriod}
+                doc={filteringDoc}
               />
-            ) : (
-              <>
-                <Divider />
-                <NoPendingReimbursements
-                  period={formattedPeriod}
-                  doc={filteringDoc}
-                />
-              </>
-            )}
-          </Section>
-          <Section>
-            <SectionTitle>{t('Reimbursements.alreadyReimbursed')}</SectionTitle>
-            {hasReimbursedExpenses ? (
-              <TransactionList
-                transactions={reimbursedExpenses}
-                className={styles.Reimbursements__transactionsList}
-                showTriggerErrors={false}
+            </>
+          )}
+        </Section>
+        <Section>
+          <SectionTitle>{t('Reimbursements.alreadyReimbursed')}</SectionTitle>
+          {hasReimbursedExpenses ? (
+            <TransactionList
+              transactions={reimbursedExpenses}
+              className={styles.Reimbursements__transactionsList}
+              showTriggerErrors={false}
+            />
+          ) : (
+            <>
+              <Divider />
+              <NoReimbursedExpenses
+                hasHealthBrands={hasHealthBrands}
+                doc={filteringDoc}
               />
-            ) : (
-              <>
-                <Divider />
-                <NoReimbursedExpenses
-                  hasHealthBrands={hasHealthBrands}
-                  doc={filteringDoc}
-                />
-              </>
-            )}
-          </Section>
-        </div>
-      </TransactionActionsProvider>
-    )
-  }
+            </>
+          )}
+        </Section>
+      </div>
+    </TransactionActionsProvider>
+  )
 }
 
-function mapStateToProps(state) {
-  return {
-    groupedExpenses: getGroupedFilteredExpenses(state),
-    currentPeriod: getPeriod(state),
-    filteringDoc: getFilteringDoc(state)
-  }
-}
+const Reimbursements = props => {
+  const { addFilterByPeriod } = useFilters()
+  const groupedExpenses = useSelector(getGroupedFilteredExpenses)
```

Comment:
i don't understand how this refacto fixes the bug.

I was expected to reuse transactions to get groupedExpenses value.

Reply:
en passant transactions en attribut ? Je suis resté sur ce qu'on s'est dit : pas la peine de faire ça puisque le hook useQuery provoque un rerender qui provoque un nouvel apprel au store via le hook useSelector, et donc un traitement avec la nouvelle donnée, non ?

Reply:
let me check on my local env

Reply:
il s'avère que le souci vient du selecteur, voir le commit dédié